src/rpc/util.cpp

```diff
@@ -101,12 +101,10 @@ CFeeRate ParseFeeRate(const UniValue& json)
 
 uint256 ParseHashV(const UniValue& v, std::string_view name)
```

Comment:
it seems to me `ParseHashV` doesn't have dedicated unit tests (only fuzz, without validation).
If you don't think it's overkill, could you please add one before this commit - to make absolutely sure that the two behaviors are the same...

(Q: what should happen when `v.isStr()` is false?)

Reply:
I'm not sure if it makes sense adding it to this branch, because I think the tests should be removed after switching `ParseHashV` to use `uint256::FromHex()` so that would probably create a bit too much unnecessary churn, but here's a commit (to which you can easily add additional inputs) you can cherry-pick that verifies the parity of both functions:

https://github.com/stickies-v/bitcoin/commit/1f2b0fa86db2ed65476b68417aa1bf4c26026a19

> (Q: what should happen when v.isStr() is false?)

I don't think this commit involves `v.isStr()`, I presume you mean `v.get_str()`? It appears to me that behaviour is unchanged, and both on master as well as in this PR a `type_error` would be thrown if `v` is not a string type?

Reply:
> I presume you mean v.get_str()

No, I meant when it doesn't actually store a string, what should be the error.

> It appears to me that behaviour is unchanged

Yes, it's just that it's a hidden error which would be eliminated if `ParseHashV` accepted a `string_view` instead of a `UniValue`. I understand if it's outside the scope, just noting that there's another path that we might want to consider when evaluating the correctness of the method.

Reply:
> No, I meant when it doesn't actually store a string, what should be the error.

Yes, that's what the `v.get_str()` call checks internally, and it throws a `type_error` if so. Have a look t the `UniValue::get_str()` implementation? (Note that user input type validation happens in `RPCHelpMan::HandleRequest()`, but of course this can still happen through internal bugs).

>  I understand if it's outside the scope

Yeah it's quite orthogonal to this PR, given the number of callsites of `ParseHashV()`, so I'm not going to change that here. I am generally in favour of pushing UniValue user input parsing into stricter types further back to the edges, though.

Reply:
> Have a look at the UniValue::get_str() implementation

I already checked all of that, that's why I was asking if we'd rather they throw on the call site. You've already answered that you think it's orthogonal.

>  I am generally in favour of pushing UniValue user input parsing into stricter types further back to the edges, though.

Yes, that's what I was saying basically - though I needed 2 rounds to understand it myself why this was bothering me.